### ISSUE TITLE
Support for 12 and 24 hour time formats

### ISSUE THREAD

**Issue by jtylerhartley** [calcom/cal.com#1990]
I am used to both the 12 hour clock rather than the 24 hour clock so I think it would be useful to provide support for both. A toggle on the settings page would be perfect. Thanks!

**Comment by PeerRich** [comment]
where do you want that setting? we already offer that in the booking page


<img width="1145" alt="image" src="https://user-images.githubusercontent.com/8019099/155892912-59fcd04b-7cfc-4527-9eae-18a5f4842e4c.png">


do you refer to the dashboard?

**Comment by jtylerhartley** [comment]
> where do you want that setting? we already offer that in the booking page
> 
> <img alt="image" width="1145" src="https://user-images.githubusercontent.com/8019099/155892912-59fcd04b-7cfc-4527-9eae-18a5f4842e4c.png">
> 
> do you refer to the dashboard?

Yes, the setting would be referencing the dashboard. I apologize for the confusion but I appreciate the opportunity to clarify things.

**Comment by PeerRich** [comment]
the dashboard should already detect your local timezone and use the right time format, does it not?

we could add a dropdown in the https://app.cal.com/settings/profile settings:

- "am/pm"
- "24 hours"

with a checkbox:

Automatically adjust time format on invitee preferences

same as we do with theme

<img width="1098" alt="image" src="https://user-images.githubusercontent.com/8019099/155894831-231646dc-0b2b-406c-94ef-3a43df09c8a3.png">

**Comment by jtylerhartley** [comment]
As shown on the image below, the availability page is displayed in a 24hr format rather than my preferred 12hr format. Changing my local time zone has no effect on the time display on the availability page in my testing so far. 

![CleanShot 2022-02-27 at 19 58 35](https://user-images.githubusercontent.com/70040274/155908187-5957a362-31d4-4ed4-96f1-c9ec9ddf1b10.png)

**Comment by baileypumfleet** [comment]
I've added support for this in PR #2002 @jtylerhartley
